pkg/api/types.go

```diff
@@ -737,6 +737,11 @@ func (a *AgentPoolProfile) HasDisks() bool {
 	return len(a.DiskSizesGB) > 0
 }
 
+// RequiresDocker returns if the kubernetes settings require docker to be installed.
+func (a *AgentPoolProfile) RequiresDocker() bool {
+	return a.KubernetesConfig.ContainerRuntime == "docker"
```

Comment:
Please forgive my ignorance, but since these are exported, is it guaranteed that this value will always be lower case during its initialization?